addons/website_event/static/src/snippets/options.js

```diff
@@ -59,14 +57,12 @@ options.registry.WebsiteEvent = options.Class.extend({
         return this._super(...arguments);
     },
     /**
+     * Ensure that we get the event object id as we could be inside a sub-object of the event
+     * like an event.track
      * @private
      */
-    _getEventObject() {
-        const repr = this.ownerDocument.documentElement.dataset.mainObject;
-        const m = repr.match(/(.+)\((\d+),(.*)\)/);
-        return {
-            model: m[1],
-            id: m[2] | 0,
-        };
+    _getEventObjectId() {
+        const objectIds = this.currentWebsiteUrl.match(/(?<=\-)[0-9]+(?!\-)(?![0-9]+)/mg);
```

Comment:
I think this should do the same. 
```suggestion
        const objectIds = this.currentWebsiteUrl.match(/(?<=-)\d+(?!-)(?!\d)/);
```
That is, 
* no need to escape `-`, 
* `[0-9]` &rarr;`\d` for 3 fewer characters, 
* if url is always single-line, ~~` m `~~
* There should be only one result, and least we only look at the first one one the next line anyways &rarr; ~~` g `~~